plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeMetastore.java

```diff
@@ -42,4 +42,6 @@ public interface DeltaLakeMetastore
     void dropTable(SchemaTableName schemaTableName, String tableLocation, boolean deleteData);
 
     void renameTable(SchemaTableName from, SchemaTableName to);
+
+    void replaceTable(Table table);
```

Comment:
We need a compatibility test with Databricks.
While on Databricks there are made operations which change the schema of the table, Trino reads & caches, if necessary the table, and we should ensure that the exception `ConcurrentModificationException` which may be appearing in `io.trino.plugin.hive.metastore.glue.GlueHiveMetastore#replaceTable` does not disrupt Trino.

It definitely shouldn't because the metadata caching operation runs in a separate thread.

Reply:
Do you have the specific suggestion of the compatibility tests? Note that this feature is enabled by default. All existing tests run with this enhancement.

Reply:
Probably my ask is not relevant.

BTW I'm not sure at the moment whether we should enable the feature by default. Maybe some customers will complain that it is too chatty for them with little apparent benefit